### PR TITLE
Map vote now cares about population at the time of the mapvote instead of round start

### PR DESCRIPTION
**Pull request by imedial**
## About The Pull Request
Mapvote will pull maps from the config based on the player count at time of vote creation, rather than round start
## Why It's Good For The Game
if there are 30 connected round start, then 80 connected when the vote is made, the options available shouldn't be the smallest possible maps for the next round
## Changelog
:cl:
fix: Map vote now cares about current player count
/:cl:

### CODE REVIEW

File: code/datums/votes/map_vote.dm [hunk 1]
```diff
@@ -10,6 +10,7 @@
 	default_choices = SSmap_vote.get_valid_map_vote_choices()
```
Comment (MrMelbert) [inline code/datums/votes/map_vote.dm:10]: Delete the one in new

File: code/datums/votes/map_vote.dm [hunk 2]
```diff
@@ -7,9 +7,9 @@
 
 /datum/vote/map_vote/New()
 	. = ..()
```
Comment (JohnFulpWillard) [inline code/datums/votes/map_vote.dm:9]: this doesn't do anything anymore so delete it

Comment (lessthnthree) [reply code/datums/votes/map_vote.dm]: This breaks map vote entirely. default_choices needs to have something in it or it won't be accessible.

```
/datum/vote/proc/is_accessible_vote()
	return !!length(default_choices)
```

image

File: code/datums/votes/map_vote.dm [hunk 3]
```diff
@@ -5,11 +5,8 @@
 	winner_method = VOTE_WINNER_METHOD_NONE
 	display_statistics = FALSE
 
-/datum/vote/map_vote/New()
-	. = ..()
-	default_choices = SSmap_vote.get_valid_map_vote_choices()
-
 /datum/vote/map_vote/create_vote()
+	default_choices = SSmap_vote.get_valid_map_vote_choices()
```
Comment (lessthnthree) [inline code/datums/votes/map_vote.dm:9]:
```suggestion
/datum/vote/map_vote/New()
	. = ..()
	default_choices = SSmap_vote.get_valid_map_vote_choices()

/datum/vote/map_vote/create_vote()
```

**Review by lessthnthree — CHANGES_REQUESTED** [review]
Set the initial default_choices in new(), ~~and refresh them at the start of can_be_initiated()~~

edit: can_be_initiated() is a bad spot

Putting the New() back in this will work for almost all map votes.
```
/datum/vote/map_vote/New()
	. = ..()
	default_choices = SSmap_vote.get_valid_map_vote_choices()
```
I was going to suggest doing a get_valid_map_vote_choices() at the start of can_be_initiated() but turns out that can_be_initiated() is called with every refresh of the TGUI votes panel. So that was less of a quick win than I was expecting.

File: code/datums/votes/map_vote.dm [hunk 4]
```diff
@@ -4,12 +4,15 @@
...
-
-/datum/vote/map_vote/New()
-	. = ..()
-	default_choices = SSmap_vote.get_valid_map_vote_choices()
+	default_choices = list(
+		"MetaStation",
+		"Birdshot Station",
+		"Ice Box Station",
+		"Tramstation",
+	)
```
Comment (MrMelbert) [inline code/datums/votes/map_vote.dm:12]: Err, if it is broken yeah revert it back to setting in `New`, don't do this

**Pull request merged**